lib/Twig/Extension/Core.php

```diff
@@ -821,7 +821,27 @@ function _twig_default_filter($value, $default = '')
 function twig_get_array_keys_filter($array)
 {
     if ($array instanceof Traversable) {
-        return array_keys(iterator_to_array($array));
+        while ($array instanceof IteratorAggregate) {
+            $array = $array->getIterator();
+        }
+
+        if ($array instanceof Iterator) {
+            $keys = array();
+            $array->rewind();
+            while ($array->valid()) {
+                $keys[] = $array->key();
+                $array->next();
+            }
+
+            return $keys;
+        }
```

Comment:
@SpacePossum Yes, actually. I rebased it against master about two weeks ago; I just needed to finish building test cases to show that there actually is a performance benefit before raising the issue again. (One of the issues I came across last time was that people seemed to think that `foreach($arr as $key => $unused)` or `foreach($arr as $key => $_)` were perfectly reasonable alternatives, which is to say, they didn't see that ignoring values was all that useful.)

Is there a specific real-world use case of this that I can look at to use as a performance example?